index.test-d.ts

```diff
@@ -1,39 +1,56 @@
 import type { ErrorObject } from "ajv";
-import { expectType } from "tsd";
+import { expectTypeOf } from "expect-type";
 import { Validator } from "./index.js";
 
 // Test static property
-expectType<Set<string>>(Validator.supportedVersions);
+expectTypeOf(Validator)
+	.toHaveProperty("supportedVersions")
+	.toEqualTypeOf<Set<string>>();
 
 // Test constructor
-expectType<Validator>(new Validator());
-expectType<Validator>(new Validator({ allErrors: true }));
+expectTypeOf(Validator).toBeConstructibleWith();
+expectTypeOf(Validator).toBeConstructibleWith({ allErrors: true });
+expectTypeOf(Validator).toBeConstructibleWith({ strict: "log" });
+expectTypeOf(Validator).toBeConstructibleWith({ strict: false });
+/** @see Validator constructor comment about ajv `strict` option */
+// @ts-expect-error
+expectTypeOf(Validator).toBeConstructibleWith({ strict: true });
+
+const someCondition: boolean = true;
+expectTypeOf(Validator).toBeConstructibleWith({
+	strict: someCondition ? "log" : false,
+});
 
 // Test instance methods
-const validator = new Validator();
-
-expectType<Promise<{ valid: boolean; errors?: ErrorObject[] | string }>>(
-	validator.validate({}),
-);
-expectType<Promise<{ valid: boolean; errors?: ErrorObject[] | string }>>(
-	validator.validate("spec.yaml"),
-);
-
-expectType<Promise<{ valid: boolean; errors?: ErrorObject[] | string }>>(
-	validator.validateBundle([{}]),
-);
-expectType<Promise<{ valid: boolean; errors?: ErrorObject[] | string }>>(
-	validator.validateBundle(["spec.yaml"]),
-);
-
-expectType<Promise<void>>(validator.addSpecRef({}, "http://example.com"));
-expectType<Promise<void>>(
-	validator.addSpecRef("spec.yaml", "http://example.com"),
-);
-
-expectType<object>(validator.resolveRefs());
-expectType<object>(validator.resolveRefs({ specification: {} }));
+interface ValidationResult {
+	valid: boolean;
+	errors?: ErrorObject[] | string;
```

Comment:
A suggestion about  array typing style: 
How about enforcing generic syntax over shorthand syntax?
Personally I found `Array<ErrorObject>` easier to read than `ErrorObject[]`

Detailed analysis can be found on this blog post:
https://tkdodo.eu/blog/array-types-in-type-script

---

To achieve this we just need to configure `useConsistentArrayType` rule

```jsonc
{
   // ...
   "useConsistentArrayType": {
     "level": "error",
     "options": {
       "syntax": "generic"
     }
   },
   // ...
}
```

Reference: https://biomejs.dev/linter/rules/use-consistent-array-type/

---

What do you think?

Reply:
Well to be honest: the difference is minimal in my view.
But I do agree that `Array<string>` reads slightly better than `string[]`. So ok to enable the extra biome rule.

Kind regards,
Hans

Reply:
Perfect, I’ve applied the change!